drivers/dma/dma_dw_common.c

```diff
@@ -546,14 +547,21 @@ int dw_dma_stop(const struct device *dev, uint32_t channel)
 	const struct dw_dma_dev_cfg *const dev_cfg = dev->config;
 	struct dw_dma_dev_data *dev_data = dev->data;
 	struct dw_dma_chan_data *chan_data = &dev_data->chan[channel];
+	enum pm_device_state pm_state;
 	int ret = 0;
 
 	if (channel >= DW_CHAN_COUNT) {
 		ret = -EINVAL;
 		goto out;
 	}
 
+	/* don't touch hw registers if not powered up */
+	ret = pm_device_state_get(dev, &pm_state);
+	if (ret != -ENOSYS && pm_state != PM_DEVICE_STATE_ACTIVE)
+		goto out;
```

Comment:
if pm_device_state_get() returns with error then we assume that the device is active?
```c
	ret = pm_device_state_get(dev, &pm_state);
	if (ret) { /* if (ret == -ENOSYS) { */
		/* Assume the device is active on error */
		pm_state = PM_DEVICE_STATE_ACTIVE;
	}

	if (pm_state != PM_DEVICE_STATE_ACTIVE ||
	    (!dw_dma_is_enabled(dev, channel) && chan_data->state != DW_DMA_SUSPENDED)) {
		goto out;
	}
```